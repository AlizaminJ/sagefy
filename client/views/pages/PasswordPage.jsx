import React from 'react'
import { shape, number, string } from 'prop-types'
import Layout from '../components/Layout'
import Icon from '../components/Icon'
import FormErrorsTop from '../components/FormErrorsTop'
import FormErrorsField from '../components/FormErrorsField'

export default function PasswordPage({ gqlErrors, state, hash }) {
  return (
    <Layout
      hash={hash}
      page="PasswordPage"
      title="Change your password"
      description="Update your Sagefy account password. Log back into your account and get learning again today."
    >
      <FormErrorsTop formErrors={gqlErrors} />
      <FormErrorsField formErrors={gqlErrors} field="all" />

      <section>
        <h1>
          Change your password <Icon i="password" s="xxl" />
        </h1>

        <ol>
          {[
            { name: 'Enter your email', icon: 'email' },
            { name: 'Check your inbox', icon: 'inbox' },
            { name: 'Change your password', icon: 'password' },
          ].map(({ name, icon }, index) => (
            <li key={`PasswordPage-steps-${name}`}>
              {index === state ? (
                <strong>
                  {name} <Icon i={icon} />
                </strong>
              ) : (
                <span>
                  {name} <Icon i={icon} />
                </span>
              )}
            </li>
          ))}
        </ol>

        {state === 0 && (
          <form action="" method="POST">
            <input type="hidden" name="state" value="0" />
            <p>
              <label htmlFor="email">Email</label>
              <input
                id="email"
                name="email"
                placeholder="example: unicorn@example.com"
                type="text"
                size="40"
                required
                autoFocus
              />
              <br />
              <small>We need your email to send the token.</small>
            </p>
            <FormErrorsField formErrors={gqlErrors} field="email" />
            <p>
              <button type="submit">
                <Icon i="password" /> Send Token
              </button>
            </p>
          </form>
        )}

        {state === 1 && <p>Please check your email inbox.</p>}

        {state === 2 && (
          <form action="" method="POST">
            <input type="hidden" name="state" value="2" />
            <p>
              <label htmlFor="newPassword">Password</label>
              <input
                id="newPassword"
                name="newPassword"
                type="password"
                size="40"
                required
                autoFocus
                pattern=".{8,}"
              />
            </p>
            <FormErrorsField formErrors={gqlErrors} field="newPassword" />
            <p>
              <button type="submit">
                <Icon i="password" /> Update Password
              </button>
            </p>
          </form>
        )}
      </section>
    </Layout>
  )
}

PasswordPage.propTypes = {
  hash: string.isRequired,
  gqlErrors: shape({}),
  state: number,
}

PasswordPage.defaultProps = {
  gqlErrors: {},
  state: 0,
}
