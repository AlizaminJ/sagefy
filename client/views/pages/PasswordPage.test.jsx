import PasswordPage from './PasswordPage'

describe('PasswordPage', () => {
  it('should render the PasswordPage 0', () => {
    expect(PasswordPage({ gqlErrors: {}, state: 0 })).toMatchSnapshot()
  })

  it('should render the PasswordPage 1', () => {
    expect(PasswordPage({ gqlErrors: {}, state: 1 })).toMatchSnapshot()
  })

  it('should render the PasswordPage 2', () => {
    expect(PasswordPage({ gqlErrors: {}, state: 2 })).toMatchSnapshot()
  })
})
