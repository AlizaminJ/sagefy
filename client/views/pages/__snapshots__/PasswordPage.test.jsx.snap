// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordPage should render the PasswordPage 0 1`] = `
<div
  className="PasswordPage"
>
  <FormErrorsTop
    formErrors={Object {}}
  />
  <FormErrorsField
    field="all"
    formErrors={Object {}}
  />
  <section>
    <h1>
      Change your password 
      <Icon
        i="password"
        s="xxl"
      />
    </h1>
    <ol>
      <li>
        <strong>
          Enter your email
           
          <Icon
            i="email"
            s="m"
          />
        </strong>
      </li>
      <li>
        <span>
          Check your inbox
           
          <Icon
            i="inbox"
            s="m"
          />
        </span>
      </li>
      <li>
        <span>
          Change your password
           
          <Icon
            i="password"
            s="m"
          />
        </span>
      </li>
    </ol>
    <form
      action=""
      method="POST"
    >
      <input
        name="state"
        type="hidden"
        value="0"
      />
      <p>
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          autoFocus={true}
          id="email"
          name="email"
          placeholder="example: unicorn@example.com"
          required={true}
          size="40"
          type="text"
        />
        <br />
        <small>
          We need your email to send the token.
        </small>
      </p>
      <FormErrorsField
        field="email"
        formErrors={Object {}}
      />
      <p>
        <button
          type="submit"
        >
          <Icon
            i="password"
            s="m"
          />
           Send Token
        </button>
      </p>
    </form>
  </section>
</div>
`;

exports[`PasswordPage should render the PasswordPage 1 1`] = `
<div
  className="PasswordPage"
>
  <FormErrorsTop
    formErrors={Object {}}
  />
  <FormErrorsField
    field="all"
    formErrors={Object {}}
  />
  <section>
    <h1>
      Change your password 
      <Icon
        i="password"
        s="xxl"
      />
    </h1>
    <ol>
      <li>
        <span>
          Enter your email
           
          <Icon
            i="email"
            s="m"
          />
        </span>
      </li>
      <li>
        <strong>
          Check your inbox
           
          <Icon
            i="inbox"
            s="m"
          />
        </strong>
      </li>
      <li>
        <span>
          Change your password
           
          <Icon
            i="password"
            s="m"
          />
        </span>
      </li>
    </ol>
    <p>
      Please check your email inbox.
    </p>
  </section>
</div>
`;

exports[`PasswordPage should render the PasswordPage 2 1`] = `
<div
  className="PasswordPage"
>
  <FormErrorsTop
    formErrors={Object {}}
  />
  <FormErrorsField
    field="all"
    formErrors={Object {}}
  />
  <section>
    <h1>
      Change your password 
      <Icon
        i="password"
        s="xxl"
      />
    </h1>
    <ol>
      <li>
        <span>
          Enter your email
           
          <Icon
            i="email"
            s="m"
          />
        </span>
      </li>
      <li>
        <span>
          Check your inbox
           
          <Icon
            i="inbox"
            s="m"
          />
        </span>
      </li>
      <li>
        <strong>
          Change your password
           
          <Icon
            i="password"
            s="m"
          />
        </strong>
      </li>
    </ol>
    <form
      action=""
      method="POST"
    >
      <input
        name="state"
        type="hidden"
        value="2"
      />
      <p>
        <label
          htmlFor="newPassword"
        >
          Password
        </label>
        <input
          autoFocus={true}
          id="newPassword"
          name="newPassword"
          pattern=".{8,}"
          required={true}
          size="40"
          type="password"
        />
      </p>
      <FormErrorsField
        field="newPassword"
        formErrors={Object {}}
      />
      <p>
        <button
          type="submit"
        >
          <Icon
            i="password"
            s="m"
          />
           Update Password
        </button>
      </p>
    </form>
  </section>
</div>
`;
